_aggregations/bucket/composite.md

```diff
@@ -0,0 +1,825 @@
+          },
+          "doc_count": 371,
+          "avg_spend": {
+            "value": 79.72514108827494
+          }
+        },
+        {
+          "key": {
+            "weekday": "Monday",
+            "gender": "FEMALE"
+          },
+          "doc_count": 320,
+          "avg_spend": {
+            "value": 72.1588623046875
+          }
+        },
+        {
+          "key": {
+            "weekday": "Monday",
+            "gender": "MALE"
+          },
+          "doc_count": 259,
+          "avg_spend": {
+            "value": 86.1754946911197
+          }
+        },
+        {
+          "key": {
+            "weekday": "Saturday",
+            "gender": "FEMALE"
+          },
+          "doc_count": 365,
+          "avg_spend": {
+            "value": 73.53236301369863
+          }
+        },
+        {
+          "key": {
+            "weekday": "Saturday",
+            "gender": "MALE"
+          },
+          "doc_count": 371,
+          "avg_spend": {
+            "value": 72.78092360175202
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
+## Paginating composite results
+
+If a request results in more than `size` buckets, then `size` buckets are returned. In this case, the result contains an `after_key` object containing the key of the next bucket in the list. To retrieve the next `size` buckets of the request, make the request again, supplying the `after_key` in the `after` parameter. For an example, see the request in [Histogram](#histogram).
+
+Always use the `after_key` rather than copying the last bucket to continue a paginated response. The two are sometimes not the same.
+{: .important}
+
+## Improving performance with index sorting
+
+To speed up composite aggregations on large datasets, you can sort your index using the same fields and order as your aggregation sources. When the `index.sort.field` and `index.sort.order` match the source fields and their order in the composite aggregation, OpenSearch can return results more efficiently and with less memory usage. While index sorting adds minor overhead during indexing, the query performance gains for composite aggregations are significant.
+
+The following example request sets sort fields and a sort order for each of the fields on the `my-sorted-index` index:
+
+```json
+PUT /my-sorted-index
+{
+  "settings": {
+    "index": {
+      "sort.field": ["customer_id", "timestamp"],
+      "sort.order": ["asc", "desc"]
+    }
+  },
+  "mappings": {
+    "properties": {
+      "customer_id": {
+        "type": "keyword"
+      },
+      "timestamp": {
+        "type": "date"
+      },
+      "price": {
+        "type": "double"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The following request creates a composite aggregation on the `my-sorted-index` index. Because the index is sorted on `customer_id` ascending and `timestamp` descending, and the aggregation sources match that sort order, this query runs faster and with reduced memory pressure:
```

Comment:
```suggestion
The following request creates a composite aggregation on the `my-sorted-index` index. Because the index is sorted by `customer_id` in ascending order and `timestamp` in descending order and the aggregation sources match that sort order, this query runs faster and with reduced memory pressure:
```